### PR TITLE
Add Sentry for exception reporting

### PR DESCRIPTION
**Pull request by achiurizo**
:clipboard: #766 

## Overview

Adds Sentry to `elixir-omg`.

## Changes

* Add [Phoenix reporting](https://github.com/getsentry/sentry-elixir#setup-with-plug-or-phoenix)
* Add [crashed process exception reporting](https://github.com/getsentry/sentry-elixir#capture-crashed-process-exceptions)
* Add [source code reporting](https://github.com/getsentry/sentry-elixir#reporting-exceptions-with-source-code)

## Testing

Need help to test it further. So far used Sentry's [included mix task](https://github.com/getsentry/sentry-elixir#testing-your-configuration) to test configuration:

```bash
SENTRY_DSN=whatever mix sentry.send_test_event
```

### CODE REVIEW

File: apps/omg/lib/omg/application.ex [hunk 1]
```diff
@@ -20,9 +20,11 @@ defmodule OMG.Application do
 
   use Application
   alias OMG.Alert.AlarmHandler
+  require Logger
 
   def start(_type, _args) do
     :ok = AlarmHandler.install()
+    {_, _} = Logger.add_backend(Sentry.LoggerBackend)
```
Comment (achiurizo) [inline apps/omg/lib/omg/application.ex]: Not sure how to do this the right way but the [documentation](https://github.com/getsentry/sentry-elixir#capture-crashed-process-exceptions) and [an issue](https://github.com/getsentry/sentry-elixir/issues/345) shows that it's only needed to be loaded once.

Comment (achiurizo) [inline apps/omg/lib/omg/application.ex]: Not sure what the best way to do this is but their documentation suggests that this only needs to be loaded once otherwise it will return an error.

File: Dockerfile [hunk 2]
```diff
@@ -70,6 +70,7 @@ RUN mix deps.clean --all
 RUN mix deps.get
 
 RUN mix compile
+RUN mix sentry.recompile
```
Comment (InoMurko) [inline Dockerfile]: why is this needed?

File: apps/omg/lib/omg/application.ex (hunk 1 above)
Comment (InoMurko) [reply apps/omg/lib/omg/application.ex]: Once per node?

File: Dockerfile (hunk 2 above)
Comment (achiurizo) [reply Dockerfile]: @InoMurko https://github.com/getsentry/sentry-elixir#reporting-exceptions-with-source-code

File: apps/omg/lib/omg/application.ex (hunk 1 above)
Comment (achiurizo) [reply apps/omg/lib/omg/application.ex]: @InoMurko once per application instance yes

Comment (InoMurko) [reply apps/omg/lib/omg/application.ex]: Hmmm. That doesn’t make too much sense. Are you sure? Where did you notice that?
Logger is elixir built in logging library that supports multiple backends. But they normally don’t need to be installed moe then once.

Comment (achiurizo) [reply apps/omg/lib/omg/application.ex]: @InoMurko I don't think I explained myself well here. This line needs to only be included once, but I am unsure where is a good place within our (umbrella?) project cc https://github.com/getsentry/sentry-elixir/issues/345 for where I was looking for guidance.

File: apps/omg_watcher_rpc/lib/application.ex [hunk 3]
```diff
@@ -50,6 +50,8 @@ defmodule OMG.WatcherRPC.Application do
       name: OMG.WatcherRPC.RootSupervisor
     ]
 
+    {_, _} = Logger.add_backend(Sentry.LoggerBackend)
```
Comment (chrishunt) [inline apps/omg_watcher_rpc/lib/application.ex]: Does it make sense to match `{:ok, _}` on these? If not, should we handle errors?

Comment (InoMurko) [reply apps/omg_watcher_rpc/lib/application.ex]: good point, definitely matching on `:ok` makes sense!  handling errors not so, if it doesn't work there's not much we can do and we want to fail completely.

Comment (achiurizo) [reply apps/omg_watcher_rpc/lib/application.ex]: Yep agreed. The reason why I went about this route is I wasn't entirely sure where the best place to put this code so and did a general match instead so I could place it everywhere and not have it complain when I loaded it more than once. 🤕  If you all have suggestions where I should load this that would be great, otherwise I'll try to create another `application.ex` for the umbrella app. re: https://github.com/getsentry/sentry-elixir/issues/345.

File: apps/omg/lib/omg/application.ex (hunk 1 above)
Comment (InoMurko) [reply apps/omg/lib/omg/application.ex]: Aha! Alright, makes sense now. If you want to understand how umbrella applications are being divided between release modes (_wathcer_, _child_chain_) you may want to look at rel/config.exs. That would explain which umbrella applications are included in one or the other modus operandi. I suggest you either include in only in _omg_, or in both `omg_watcher_rpc` and `omg_child_chain_rpc`.

File: dialyzer.ignore-warnings [hunk 4]
```diff
@@ -5,6 +5,8 @@ apps/omg_eth/test
 
 # TODO
 
+Unknown function 'Elixir.Sentry':capture_exception/2
```
Comment (InoMurko) [inline dialyzer.ignore-warnings]: If you add `:sentry` in the PLT app list in `mix.exs` you don't need to add these ignore here.
But there's an open issues that will produce an umatched_return: https://github.com/getsentry/sentry-elixir/pull/344/files


Also, I think you forgot 
```
use Plug.ErrorHandler
use Sentry.Plug
```
to add in the integration.

File: apps/omg/lib/omg/application.ex [hunk 5]
```diff
@@ -20,9 +20,11 @@ defmodule OMG.Application do
 
   use Application
   alias OMG.Alert.AlarmHandler
+  require Logger
 
   def start(_type, _args) do
     :ok = AlarmHandler.install()
+    {:ok, _} = Logger.add_backend(Sentry.LoggerBackend)
```
Comment (pthomalla) [inline apps/omg/lib/omg/application.ex]: move this option to `config/config.exs`
```elixir
 config :logger,
    backends: [Sentry.LoggerBackend, :console]
```
tests will pass, and backend will be available even application is restarted.

File: apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/application.ex [hunk 6]
```diff
@@ -32,6 +32,7 @@ defmodule OMG.ChildChainRPC.Application do
       {OMG.ChildChainRPC.Web.Endpoint, []}
     ]
 
+
```
Comment (pthomalla) [inline apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/application.ex]: unnecessary enter

File: apps/omg_status/lib/omg_status/application.ex [hunk 7]
```diff
@@ -31,6 +31,7 @@ defmodule OMG.Status.Application do
         []
       end
 
+
```
Comment (pthomalla) [inline apps/omg_status/lib/omg_status/application.ex]: unnecessary enter

File: apps/omg_watcher_rpc/lib/application.ex [hunk 8]
```diff
@@ -50,6 +50,7 @@ defmodule OMG.WatcherRPC.Application do
       name: OMG.WatcherRPC.RootSupervisor
     ]
 
+
```
Comment (pthomalla) [inline apps/omg_watcher_rpc/lib/application.ex]: unnecessary enter

File: config/config.exs [hunk 9]
```diff
@@ -18,6 +18,21 @@ config :logger, :console,
   discard_threshold: 2000,
   metadata: [:module, :function, :request_id]
 
+config :logger,
+  backends: [Sentry.LoggerBackend, :console]
+
+config :sentry,
```
Comment (InoMurko) [inline config/config.exs]: Just remembered... have you found any docs how Sentry works with a elixir release?

File: apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/web/endpoint.ex [hunk 10]
```diff
@@ -15,6 +15,7 @@
 defmodule OMG.ChildChainRPC.Web.Endpoint do
   use Phoenix.Endpoint, otp_app: :omg_child_chain_rpc
   use Appsignal.Phoenix
+  use Sentry.Phoenix.Endpoint
```
Comment (pthomalla) [inline apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/web/endpoint.ex]: add the same line to the watcherapps/omg_watcher_rpc/lib/web/endpoint.ex

File: config/config.exs (hunk 9 above)
Comment (achiurizo) [reply config/config.exs]: @InoMurko nope I haven't looked for any so far. Can follow up with another PR to improve this integration.
